src/Text/Pandoc/Writers/HTML.hs

```diff
@@ -1022,11 +1022,13 @@ inlineToHtml opts inline = do
 
     (Emph lst)       -> inlineListToHtml opts lst >>= return . H.em
     (Strong lst)     -> inlineListToHtml opts lst >>= return . H.strong
-    (Code attr str)  -> case hlCode of
+    (Code attr@(ids,classes,kvs) str)  
+                     -> case hlCode of
                              Left msg -> do
                                unless (null msg) $
                                  report $ CouldNotHighlight msg
-                               addAttrs opts attr $ H.code $ strToHtml str
+                               addAttrs opts (ids,classes',kvs) $ 
+                                 elemType $ strToHtml str
                              Right h -> do
                                modify $ \st -> st{ stHighlighting = True }
                                addAttrs opts (id',[],keyvals) h
```

Comment:
I don't understand what needs to be done for this case.

Reply:
Sorry, I'm not sure I understand your question - can you elaborate?

Reply:
When `hlCode` is a `Left` value, it was straightforward to choose from `H.code`, `H.var`, or `H.samp`. But when `isJust (writerHighlightStyle opts)` is `True` (i.e. `hlCode` is a `Right` value), the `Html` is generated from ``` highlight (writerSyntaxMap opts) formatHtmlInline attr str ``` is output. Do we need to support highlighted `<samp>` or `<var>`?

Reply:
Ah, I see.  The highlighted code returned by skylighting already includes a code tag, right?
Well, maybe we could just include the whole thing under `samp`.  As far as I can see, all phrasing content, including `<code>`, is legal inside a `<samp>` element. Same for `<var>`.

Reply:
Added `<samp>` and `<var>` support for highlighted code.